lib/dor/text_extraction/workflow_updater.rb

```diff
@@ -2,25 +2,36 @@
 
 module Dor
   module TextExtraction
-    # Update the status of workflow steps in SDR based on OCR processing events
+    # Update the status of workflow steps in SDR based on text extraction (OCR, speech-to-text) processing events
     class WorkflowUpdater
-      attr_reader :workflow, :step
+      OCR_WF_NAME = 'ocrWF'
+      OCR_WF_CREATE_NAME = 'ocr-create'
+      STT_WF_NAME = 'speechToTextWF'
+      STT_WF_CREATE_NAME = 'stt-create'
 
       # Default is to update the 'ocr-create' step in the 'ocrWF' workflow
-      def initialize(workflow: 'ocrWF', step: 'ocr-create', client: nil, logger: nil)
-        @workflow = workflow
-        @step = step
+      def initialize(client: nil, logger: nil)
         @client = client || LyberCore::WorkflowClientFactory.build(logger:)
       end
 
       # Notify SDR that the OCR workflow step completed successfully
-      def mark_ocr_completed(druid)
-        @client.update_status(druid:, workflow:, process: step, status: 'completed')
+      def mark_ocr_create_completed(druid)
+        @client.update_status(druid:, workflow: OCR_WF_NAME, process: OCR_WF_CREATE_NAME, status: 'completed')
```

Comment:
hmm, hadn't thought about that, but thinking it through now...

1. based on a quick read of the workflow-server-rails controller code for `update` (pretty cursory, so not sure of my interpretation): it seems like [current status has to be passed in the params to get a `conflict` error if the status has changed since the update call was made](https://github.com/sul-dlss/workflow-server-rails/blob/4f102d441e525ef42b2cb99bac4b01683852625a/app/controllers/steps_controller.rb#L46).  but we're not doing that here, so i think it'll just get marked `completed`?  i guess that could be problematic since that `update` operation also queues the next step, which would already have been queued if status had already been set.
2. but i think that we'd need to rework the workflow-server-rails controller method a bit to totally prevent race conditions.  if we check the status here before trying to update, we're not guaranteed to avoid the race condition, since the check and update are separate REST calls that can't be wrapped in a transaction.  but even though it'd be a much more fleeting window, there's still an opportunity (afaict) for a race condition on the workflow-server-rails side, because the query result against withc the `current-status` param value is checked isn't obtained in the same transaction in which the update happens.

so... my gut feeling is we should just try our best to avoid this possibility by tuning the visibility timeout?  currently the only thing that uses the SQS poller is `speech_to_text_watcher`, with the `SqsWatcher` default of 120 seconds, and the only thing the `speech_to_text_watcher` has the poller do is update a workflow event and send a DSA event, currently with no retries.  so it feels like that should succeed or fail well within 2 min.  but i'd be happy to just turn it up to 10 or something to be safe.

thinking about it a bit more, i think our latent assumption in most of the SDR queues is that a particular job only gets picked up once, and we're not always great about making job invocations idempotent.  but to be fair, sidekiq superfetch (sidekiq in general?) just guarantees that a job will be run at least once, not exactly once.  so there are weird job timeout situations where things can get double run in other parts of SDR (tho i think that's effectively even more of a corner case than the situation here, and we probably wouldn't run into that sort of thing outside of the sidekiq hotswap timeout being hit for a job spanning a deploy, or some sort of weird redis/worker networking issue between when a job is picked up and when it'd be marked done).

Reply:
My contribution to this process is to agree that the `update` operation will happily do nothing if you don't pass in the `current-status`.  But also that if the step gets marked as completed before it's actually done, the workflow will proceed, perhaps incorrectly.